ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -465,7 +465,9 @@ private void processReplicaMetadataResponse(Set<StoreKey> missingStoreKeys,
           MessageFormatInputStream deleteStream =
               new DeleteMessageFormatInputStream(messageInfo.getStoreKey(), messageInfo.getAccountId(),
                   messageInfo.getContainerId(), messageInfo.getOperationTimeMs());
-          MessageInfo info = new MessageInfo(messageInfo.getStoreKey(), deleteStream.getSize(), true);
+          MessageInfo info =
```

Comment:
Is there a reason why we generate a new MessageInfo here? `deleted` would already be set to true from the if condition above, right?